src/discretization/point_generators.jl

```diff
@@ -120,3 +121,366 @@ function uniform_sphere(diameter::Real, ΔX0::Real; center_x::Real=0, center_y::
     isapprox(center_z, 0; atol=eps()) || (position[3, :] .+= center_z)
     return position, volume
 end
+
+"""
+    uniform_cylinder(diameter::Real, height::Real, ΔX0::Real; kwargs...)
+
+Creates a grid of uniformly distributed points in a cylindrical shape with a specific
+`height` in z-dimension, a `diameter` and the point spacing `ΔX0`. Due to the uniform point
+spacings, edges on the surface of the cylinder can occur.
+
+# Arguments
+- `diameter::Real`: Diameter of the cylinder.
+- `height::Real`: Height of the cylinder.
+- `ΔX0::Real`: Spacing of the points.
+
+# Keywords
+- `center_position`: The coordinates of the center of the cylinder. Default: `(0, 0, 0)`
+
+# Returns
+- `position::Matrix{Float64}`: A `3×n_points` matrix with the position of the points.
+- `volume::Vector{Float64}`: A vector with the volume of each point.
+
+# Examples
+
+```julia-repl
+julia> position, volume = uniform_cylinder(5, 10, 2);
+
+julia> position
+3×20 Matrix{Float64}:
+ -1.0   1.0  -1.0   1.0  -1.0   1.0  -1.0  …   1.0  -1.0  1.0  -1.0   1.0  -1.0  1.0
+ -1.0  -1.0   1.0   1.0  -1.0  -1.0   1.0     -1.0   1.0  1.0  -1.0  -1.0   1.0  1.0
+ -4.0  -4.0  -4.0  -4.0  -2.0  -2.0  -2.0      2.0   2.0  2.0   4.0   4.0   4.0  4.0
+
+julia> volume
+20-element Vector{Int64}:
+ 8
+ 8
+ 8
+ 8
+ 8
+ ⋮
+ 8
+ 8
+ 8
+ 8
+ 8
+```
+"""
+function uniform_cylinder(diameter::Real, height::Real, ΔX0::Real; center_position=(0, 0, 0))
+radius = diameter / 2
+center_x, center_y, center_z = center_position
+_gridxy = range(- radius + ΔX0 / 2, radius - ΔX0 / 2; step=ΔX0)
+gridxy = _gridxy .- sum(_gridxy) / length(_gridxy)
+_gridz = range((-height + ΔX0) / 2, (height - ΔX0) / 2; step=ΔX0)
+gridz = _gridz .- sum(_gridz) / length(_gridz)
+__position = vec(collect(Iterators.product(gridxy, gridxy, gridz)))
+_position = reinterpret(reshape, eltype(eltype(__position)), __position)
+cylinder_points = find_points(p -> √(p[1]^2 + p[2]^2) ≤ radius, _position)
+position = _position[:, cylinder_points]
+volume = fill(ΔX0^3, size(position, 2))
+isapprox(center_x, 0; atol=eps()) || (position[1, :] .+= center_x)
+isapprox(center_y, 0; atol=eps()) || (position[2, :] .+= center_y)
+isapprox(center_z, 0; atol=eps()) || (position[3, :] .+= center_z)
+return position, volume
+end
```

Comment:
This should be formatted correctly:
```suggestion
function uniform_cylinder(diameter::Real, height::Real, ΔX0::Real;
                          center_position=(0, 0, 0))
    radius = diameter / 2
    center_x, center_y, center_z = center_position
    _gridxy = range(-radius + ΔX0 / 2, radius - ΔX0 / 2; step=ΔX0)
    gridxy = _gridxy .- sum(_gridxy) / length(_gridxy)
    _gridz = range((-height + ΔX0) / 2, (height - ΔX0) / 2; step=ΔX0)
    gridz = _gridz .- sum(_gridz) / length(_gridz)
    __position = vec(collect(Iterators.product(gridxy, gridxy, gridz)))
    _position = reinterpret(reshape, eltype(eltype(__position)), __position)
    cylinder_points = find_points(p -> √(p[1]^2 + p[2]^2) ≤ radius, _position)
    position = _position[:, cylinder_points]
    volume = fill(ΔX0^3, size(position, 2))
    isapprox(center_x, 0; atol=eps()) || (position[1, :] .+= center_x)
    isapprox(center_y, 0; atol=eps()) || (position[2, :] .+= center_y)
    isapprox(center_z, 0; atol=eps()) || (position[3, :] .+= center_z)
    return position, volume
end
```